system/Bootstrap.cfc

```diff
@@ -1,719 +1,795 @@
-	 * @COLDBOX_APP_KEY The key used in application scope for this application
-	 * @COLDBOX_APP_MAPPING The application mapping override, only used for Flex/SOAP apps, this is auto-calculated
-	 * @COLDBOX_FAIL_FAST By default if an app is reiniting and a request hits it, we will fail fast with a message. This can be a boolean indicator or a closure.
-	 */
-	Bootstrap function init(
-		required string COLDBOX_CONFIG_FILE,
-		required string COLDBOX_APP_ROOT_PATH,
-		string COLDBOX_APP_KEY,
-		string COLDBOX_APP_MAPPING="",
-		any COLDBOX_FAIL_FAST=true
-	){
-		// Set vars for two main locations
-		setCOLDBOX_CONFIG_FILE( arguments.COLDBOX_CONFIG_FILE );
-		setCOLDBOX_APP_ROOT_PATH( arguments.COLDBOX_APP_ROOT_PATH );
-		setCOLDBOX_APP_MAPPING( arguments.COLDBOX_APP_MAPPING );
-		setCOLDBOX_FAIL_FAST( arguments.COLDBOX_FAIL_FAST );
-
-		// App Key Check
-		if( structKeyExists( arguments, "COLDBOX_APP_KEY" ) AND len( trim( arguments.COLDBOX_APP_KEY ) ) ){
-			setCOLDBOX_APP_KEY( arguments.COLDBOX_APP_KEY );
-		}
-
-		return this;
-	}
-
-	/**
-	 * Loads the framework into application scope and executes app start procedures
-	 *
-	 * @throws InvalidColdBoxMapping
-	*/
-	function loadColdBox(){
-		var appKey = locateAppKey();
-
-		// Cleanup of old code, just in case
-		if( structkeyExists( application, appKey ) ){
-			structDelete( application, appKey );
-		}
-
-		// Verify Mapping
-		if( !fileExists( expandPath( "/coldbox/system/web/Controller.cfc" ) ) ){
-			var coldboxDirectory = reReplaceNoCase(
-				getDirectoryFromPath( getCurrentTemplatePath() ),
-				"[\\/]system",
-				""
-			);
-			throw(
-				message = "Cannot find the '/'coldbox' mapping",
-				detail 	= "It seems that you do not have a '/coldbox' mapping in your application and we cannot continue to process the request.
+/**
+ * Copyright Since 2005 ColdBox Framework by Luis Majano and Ortus Solutions, Corp
+ * www.ortussolutions.com
+ * ---
+ * @author Luis Majano <lmajano@ortussolutions.com>
+ * Loads the framwork into memory and provides a ColdBox application.
+ */
+component serializable="false" accessors="true" {
+
+    /************************************** CONSTRUCTOR *********************************************/
+
+    // Configuration File Override
+    property name="COLDBOX_CONFIG_FILE";
+    // Application root path
+    property name="COLDBOX_APP_ROOT_PATH";
+    // The key used in application scope to model this app
+    property name="COLDBOX_APP_KEY";
+    // The application mapping override, only used for Flex/SOAP apps, this is auto-calculated
+    property name="COLDBOX_APP_MAPPING";
+    // Lock Timeout for startup operations
+    property name="lockTimeout";
+    // The application hash used for locks
+    property name="appHash";
+    // By default if an app is reiniting and a request hits it, we will fail fast with a message
+    property name="COLDBOX_FAIL_FAST";
+
+    // param the properties with defaults
+    param name="COLDBOX_CONFIG_FILE" default="";
```

Comment:
@elpete I have some concerns with the auto-formmatting that maybe we can involve John.  I am marking them here. Basically, the auto formatting is loosing my code alignments